frontend/src/components/AddPlayForm.js

```diff
@@ -9,30 +9,44 @@ import PlayersField, { PlayersFieldValidator } from './PlayersField'
 import PlayingTimeField, { PlayingTimeFieldValidator } from './PlayingTimeField'
 import PlayRatingField from './PlayRatingField'
 import { PrimaryButton } from './Button'
+import PlayImageField, { PlayImageFieldValidator } from './PlayImageField.js'
+import { uploadImage, uploadGameData } from '../environment/playDataRestClient'
 
 const AddPlaySchema = Yup.object().shape({
+  playImage: PlayImageFieldValidator,
   gameTitle: GameTitleFieldValidator,
   playDate: PlayDateFieldValidator,
   players: PlayersFieldValidator,
   playingTime: PlayingTimeFieldValidator,
 })
 
-export default function AddPlayForm( { addToPlayCollection }) {
+export default function AddPlayForm({ addToPlayCollection }) {
   return <div>
     <h1>Add a New Play to your Timeline</h1>
+    <p>* Rquired</p>
     <Formik
       initialValues={{
+        playImage: '',
         gameTitle: '',
         playDate: '',
         players: '',
         playingTime: '',
         playRating: 5
       }}
+      validateOnChange={true}
       validationSchema={AddPlaySchema}
-      onSubmit={values => addToPlayCollection(values)}     
+      onSubmit={(values, { setSubmitting, resetForm }) => {
+        uploadImage(values.playImage)
+          .then(imageURL => uploadGameData(values, imageURL))
+          .then(savedPlayValues => addToPlayCollection(savedPlayValues))
+          .then(() => setSubmitting(false))
+          .finally(() => resetForm())
```

Comment:
Wenn du eine Funktion so übergibst: `finally(() => f())` kannst du auch direkt `finally(f)` schreiben 
```suggestion
          .finally(resetForm)
```

Reply:
Danke, das habe ich vergessen.